### DISCUSSION TITLE
changing soundfont file on a channel via midi CC messages?

### DISCUSSION THREAD

**Discussion by donundeen** [FluidSynth/fluidsynth#1353] [Q&A] [answered]
Hi all, 
I'm currently using node-easymidi to send MIDI messages to fluidsynth running as a service on a raspberry pi.

I'm  loading multiple soundfont files via the 
SOUNDFONT=[file list]
line in /etc/default/fluidsynth file. 
I can also start fluidsynth interactive mode from the command-line with multiple soundfont files, no problem.

In interactive mode, I can select different soundfont files with the command
select [channel] [soundfont number] [bank] [program]

But that doesn't work through easymidi, because I think that's not a supported MIDI message?

I know I can use CC message 000 to send bank select messages, but is there a way to use CC messages to select the soundfont number?

Or is there another way to manage different soundfont files via MIDI messages to a fluidsynth running as a service? 

Thanks!

**Comment by derselbst** [comment, marked as answer]
MIDI doesn't know about soundfont and therefore no MIDI event event exists to switch between soundfonts. Your options are:

* Use fluidsynth's API directly (`fluid_synth_program_select()`)
* Use fluidsynth's sequencer, which provides custom events (similar to MIDI, yet different), which has an event type `FLUID_SEQ_PROGRAMSELECT`
* Structure the presets of your soundfonts in such a way, that you can use bank select messages (CC00 and CC32) to switch between soundfonts, by exploiting [fluidsynth's soundfont stack](https://stackoverflow.com/a/53307571)

**Comment by unhandyandy** [comment]
Thanks, I've successfully solved this issue with your 3rd suggestion.  I used Polyphone to easily change the bank numbers in my favorite sf2 files to avoid conflicts.
